Content.Server/Singularity/EntitySystems/SingularitySystem.cs

```diff
@@ -21,56 +37,157 @@ public override void Initialize()
             if (component.BeingDeletedByAnotherSingularity)
                 return;
 
-            var otherEntity = args.OtherFixture.Body.Owner;
+            // Using this to also get smooth deletions is hard because we need to be hard for good bounce
+            // off of containment but also we need to be non-hard so we can freely move through the station.
+            // For now I've just made it so only the lookup does deletions and collision is just for fields.
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+            _gravityAccumulator += frameTime;
+            _accumulator += frameTime;
 
-            if (otherEntity.TryGetComponent<IMapGridComponent>(out var mapGridComponent))
+            while (_accumulator > _updateInterval)
             {
-                foreach (var tile in mapGridComponent.Grid.GetTilesIntersecting(args.OurFixture.Body.GetWorldAABB()))
+                _accumulator -= _updateInterval;
+
+                foreach (var singularity in ComponentManager.EntityQuery<ServerSingularityComponent>())
                 {
-                    mapGridComponent.Grid.SetTile(tile.GridIndices, Robust.Shared.Map.Tile.Empty);
-                    component.Energy++;
+                    singularity.Energy -= singularity.EnergyDrain;
                 }
-                return;
             }
 
-            if (otherEntity.HasComponent<ContainmentFieldComponent>() ||
-                (otherEntity.TryGetComponent<ContainmentFieldGeneratorComponent>(out var containmentField) && containmentField.CanRepell(component.Owner)))
+            while (_gravityAccumulator > GravityCooldown)
             {
-                return;
+                _gravityAccumulator -= GravityCooldown;
+
+                foreach (var singularity in ComponentManager.EntityQuery<ServerSingularityComponent>())
+                {
+                    Update(singularity, GravityCooldown);
+                }
             }
+        }
 
-            if (otherEntity.IsInContainer())
-                return;
+        private void Update(ServerSingularityComponent component, float frameTime)
+        {
+            if (component.BeingDeletedByAnotherSingularity) return;
+
+            var worldPos = component.Owner.Transform.WorldPosition;
+            DestroyEntities(component, worldPos);
+            DestroyTiles(component, worldPos);
+            PullEntities(component, worldPos);
+        }
+
+        private float PullRange(ServerSingularityComponent component)
+        {
+            // Level 6 is normally 15 range but that's yuge.
+            return 2 + component.Level * 2;
+        }
+
+        private float DestroyTileRange(ServerSingularityComponent component)
+        {
+            return component.Level - 0.5f;
+        }
+
+        private void HandleDestroy(ServerSingularityComponent component, IEntity entity)
+        {
+            // TODO: Need singuloimmune tag
```

Comment:
Being singulo-eatable should probably be opt-in, not opt-out (just a note)

Reply:
The fact that 99% of entities are destroyed by the singulo I think it'd be more annoying to make it opt-in, especially when you'd have to add a tagcomponent to effectively everything.